src/esp/bindings_js/bindings_js.cpp

```diff
@@ -132,16 +132,11 @@ bool isBuildWithBulletPhysics() {
 #endif
 }
 
-int castRay(Simulator& sim,
-            const Magnum::Vector3& pos,
-            const Magnum::Vector3& dir,
-            float maxDistance) {
-  esp::geo::Ray ray;
-  ray.origin = pos;
-  ray.direction = dir;
-  auto result = sim.castRay(ray, maxDistance, 0);
-  // note that hits[0].objectId may be -1 to indicate hit on stage
-  return result.hasHits() ? result.hits[0].objectId : -1;
+RaycastResults castRay(Simulator& sim,
+                       const Ray& ray,
+                       float maxDistance,
+                       int sceneID) {
+  return sim.castRay(ray, maxDistance, sceneID);
```

Comment:
Any reason you don't want top bind this function to the Simulator class?

Reply:
Fixed, thanks!